src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarcStartup.cs

```diff
@@ -31,14 +37,31 @@ public ProcessHostMonarcStartup(ILoggerFactory loggerFactory, ILogger<ProcessHos
             _downloader = downloader;
             _lifetime = lifetime;
             _launcherExecutablePath = launcherExecutablePath;
+            _kestrelReady = new TaskCompletionSource<Uri>();
+
+            hostApplicationLifetime.ApplicationStarted.Register(() =>
+            {
+                try
+                {
+                    _kestrelReady.TrySetResult(new Uri(server.Features.Get<IServerAddressesFeature>()!.Addresses!.First()));
```

Comment:
Does kestrel select the next available port? And how do we get this info in the downstream processes?

Reply:
if the launcher port is 0 then kestrel requests a free port from the OS https://learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-7.0#port-0.

Later we update that value in the launcher configuration. I've moved where we do that to the begining of the ExecuteAsync method so its more clear whats going on.